File: backend/src/main/java/com/backend/domain/repository/dto/response/github/IssueResponse.java
package com.backend.domain.repository.dto.response.github;

// issue 응답용 DTO
public record IssueResponse(
        Long number,
        String title,
        String state,
        String created_at,
        String closed_at,
        PullRequest pull_request
) {
    public record PullRequest(String url) {}

    public boolean isPureIssue() {
        return pull_request == null;
    }

    public boolean isClosed() {
        return "closed".equals(state);
    }
}

File: backend/src/main/java/com/backend/domain/repository/dto/response/github/IssueResponse.kt
package com.backend.domain.repository.dto.response.github

// issue 응답용 DTO
data class IssueResponse(
    val number: Long?,
    val title: String?,
    val state: String?,
    val created_at: String?,
    val closed_at: String?,
    val pull_request: PullRequest?
) {
    data class PullRequest(val url: String?)

    val isPureIssue: Boolean
        get() = pull_request == null

    val isClosed: Boolean
        get() = state == "closed"
}
